### ISSUE TITLE
Rule proposal: option for `prefer-dom-node-append`

### ISSUE THREAD

**Issue by liangjiancang** [sindresorhus/eslint-plugin-unicorn#1517]
`Node#append()` is not always better than `Node#appendChild()` because the latter can return the added node. An option is needed to allow for this.

## Fail

```js
parent.appendChild(document.createElement('div'))
```

## Pass

```js
const child = parent.appendChild(document.createElement('div'))
fn(parent.appendChild(document.createElement('div')))
return parent.appendChild(document.createElement('div'))
```

**Comment by sindresorhus** [comment]
You can just keep a reference to the node and return that:

```js
const foo = document.createElement('div');
parent.append(foo);
return foo;
```

**Comment by liangjiancang** [comment]
Yes, new code can all be written in this form. But there are probably thousands of `appendChild` in an existed project, and projects like this have to reject the rule `prefer-dom-node-append`.

**Comment by sindresorhus** [comment]
But the point of the rule is to use `.append()` everywhere possible. If you cannot do that, you can just disable the rule and slowly migrate.
